sailthru_content/services/catalog_api_service.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from edx_rest_api_client.client import EdxRestApiClient
+
+
+logger = logging.getLogger()
+COURSES_PAGE_SIZE = 500
+PROGRAMS_PAGE_SIZE = 40
+
+
+class CatalogApiService(object):
+
+    _cached_programs = {}
```

Comment:
This is not a django application. This is a script. So in a script setting, the data saved into the attribute is "cached" for the next call. I can certainly think of another name for this.